persistent-mysql/Database/Persist/MySQL.hs

```diff
@@ -1034,10 +1037,25 @@ insertOnDuplicateKeyUpdate
 insertOnDuplicateKeyUpdate record =
   insertManyOnDuplicateKeyUpdate [record] []
 
--- | This wraps values of an Entity's 'EntityField', making them have the same
--- type. This allows them to be put in lists.
+-- | This type is used to determine how to update rows using MySQL's
+-- @INSERT ON DUPLICATE KEY UPDATE@ functionality, exposed via
+-- 'insertManyOnDuplicateKeyUpdate' in the library.
 data SomeField record where
   SomeField :: EntityField record typ -> SomeField record
+  -- ^ Copy the field directly from the record.
+  CopyUnlessEq :: PersistField typ => EntityField record typ -> typ -> SomeField record
+  -- ^ Only copy the field if it is not equal to the provided value.
+
+-- | Copy the field into the database only if the value in the
```

Comment:
Can you bump the cabal version, update changelog and add the `@since` haddock syntax for all the functions you have added here.

Reply:
Sure thing! Is adding a constructor like this a minor version bump?

Reply:
You need to bump it to `2.6.2`. More info here: http://www.snoyman.com/blog/2017/06/how-to-send-me-a-pull-request